### PR TITLE
Add support for non-superuser loading of the age plugin

### PR DESCRIPTION
**Pull request by daanwa**
In relation to issue [839](https://github.com/apache/age/issues/839)

### CODE REVIEW

**Comment by daanwa** [comment]
In my initial PR I did not add the superuser parameter to the init file from which most of the time you initiate the connect.

File: drivers/python/age/__init__.py [hunk 1]
```diff
@@ -24,12 +24,13 @@ def version():
     return VERSION.VERSION
 
 
-def connect(dsn=None, graph=None, connection_factory=None, cursor_factory=ClientCursor, **kwargs):
+def connect(dsn=None, graph=None, connection_factory=None, cursor_factory=ClientCursor, superuser:bool=True, **kwargs):
```
Comment (rafsun42) [inline drivers/python/age/__init__.py:27]: To be consistent with rest of the parameters, I would avoid specifiying the type.

File: drivers/python/age/age.py [hunk 2]
```diff
@@ -39,9 +39,13 @@ def load(self, data: bytes | bytearray | memoryview) -> Any | None:
...
 
 
-def setUpAge(conn:psycopg.connection, graphName:str):
+def setUpAge(conn:psycopg.connection, graphName:str, superuser:bool=True):
     with conn.cursor() as cursor:
-        cursor.execute("LOAD 'age';")
+        if superuser:
+            cursor.execute("LOAD 'age';")
+        else:
+            cursor.execute("LOAD '$libdir/plugins/age';")
```
Comment (rafsun42) [inline drivers/python/age/age.py:47]: This assumes that the shared object is present in the `$libdir/plugins` directory. That may not be correct if DBA does not install age there.

**Comment by rafsun42** [comment]
@daanwa The python driver test did not pass.

File: drivers/python/age/age.py (hunk 2 above)
Comment (daanwa) [reply drivers/python/age/age.py:47]: I'm under the impression that while $libdir may change on the system, the "plugins" directory is somewhat static and should only contain plugins that may be used by non-super users. So if a DBA installs this in a different directory, it won't work either way. However I'm not a DBA and not fully aware of all postgres specifics. Perhaps we could leave it like this for now and see if/when this requires further configuration options.

File: drivers/python/age/__init__.py (hunk 1 above)
Comment (daanwa) [reply drivers/python/age/__init__.py:27]: Makes sense.

**Comment by daanwa** [comment]
The build fail seems to be due to some kind of timeout on files.pythonhosted.org. Not sure if there is anything I can change there.

File: drivers/python/age/age.py (hunk 2 above)
Comment (rafsun42) [reply drivers/python/age/age.py:52]: @daanwa In that case, I suggest we rename the parameter to something more descriptive, such as `in_plugins_dir` or `load_from_plugins`. Additionally, we should explain what this parameter does in the documentation.

Comment (daanwa) [reply drivers/python/age/age.py:52]: Yes, agree, that's a better way to go about it. I'll use the `load_from_plugins` name.

**Comment by rafsun42** [comment]
@daanwa Looks good. Could you please squash all the commits into one commit and add a body to the commit message briefly describing all the changes being added?

**Comment by daanwa** [comment]
I Squashed the commits for my changes into two because I unfortunatly synced the fork between several of my changes. Hopefully this will do.

**Comment by rafsun42** [comment]
@daanwa I see there is an unintended commit. Unfortunately, we cannot accept that with this PR. My suggestion is to rebase your changes to latest master. Ideally, your changes should be added to a non-master branch. It makes it easy to rebase in case master is updated before the PR is merged.

**Pull request closed without merging**